stable_baselines3/common/base_class.py

```diff
@@ -10,6 +10,7 @@
 import gym
 import numpy as np
 import torch as th
+from typing_extensions import Self
```

Comment:
typing extensions might not be installed (pytype is an optional dependency), I would also keep it optional if possible

Reply:
Yep, I think to avoid this dependency we have two options.

Simplest but somewhat ugly is just to go with the TypeVar solution described in https://peps.python.org/pep-0673/ that Self is aimed to replace. Perhaps not so bad if we only need to do this in a few places.

Alternatively, can make the import only happen for type evaluation:
```
if typing.TYPE_CHECKING:
    from typing_extensions import Self
```
Then we might just be able to write `"Self"` instead of `Self`? If string literals not supported, you could also do something janky like:
```
if typing.TYPE_CHECKING:
    from typing_extensions import Self
else:
    Self = "BaseAlgorithm"
```

Reply:
> typing extensions might not be installed

No, typing-extensions is installed, since it is a torch dependency since at least version 1.11.

Reply:
But to keep it clean, I recommend adding it to the required dependencies of SB3.